deepchem/models/torch_models/layers.py

```diff
@@ -4792,3 +4792,129 @@ def forward(self, dist: torch.Tensor) -> torch.Tensor:
         dist = dist.unsqueeze(-1) / self.cutoff
         output: torch.Tensor = self.envelope(dist) * (self.freq * dist).sin()
         return output
+
+
+class DecoderRNN(nn.Module):
+    """Decoder Layer for SeqToSeq Model.
+
+    The decoder transforms the embedding vector into the output
+    sequence. It is trained to predict the next token in the sequence given the
+    previous tokens in the sequence. It uses the context vector from the encoder
+    to help generate the correct token in the sequence.
+
+    Examples
+    --------
+    >>> from deepchem.models.torch_models.layers import DecoderRNN
+    >>> import torch
+    >>> embedding_dimensions = 5
+    >>> num_output_tokens = 7
+    >>> num_input_tokens = 12
+    >>> max_length = 4
+    >>> batch_size = 2
+    >>> device = 'cuda' if torch.cuda.is_available() else 'cpu'
+    >>> layer = DecoderRNN(embedding_dimensions, num_output_tokens,
+    ...                                 max_length, device)
+    >>> embeddings = torch.randn(batch_size, num_input_tokens, embedding_dimensions)
+    >>> output, hidden, _ = layer([embeddings,
+    ...                            embeddings[:, -1].unsqueeze(0).contiguous(), None])
+    >>> output.shape
+    torch.Size([2, 4, 7])
+
+    References
+    ----------
+    .. [1] Sutskever et al., "Sequence to Sequence Learning with Neural Networks"
+
+    """
+
+    def __init__(self,
+                 hidden_size:int,
+                 output_size:int,
+                 max_length:int,
+                 device: torch.device,
```

Comment:
try to figure out the root cause of why the devices are mismatched, rather than pass in device as an argument.

Reply:
In the forward function when we initialise a tensor it goes to the cpu.
While if we initialise the same in init function.

Reply:
nevermind i found a work around in graph_utility.py

Reply:
![Screenshot 2023-08-25 104742](https://github.com/deepchem/deepchem/assets/49408798/b993f3e4-a675-49d6-b1ef-d8a607fc0925)